acceptance/tests/windows/prevent_password_changes.rb

```diff
@@ -0,0 +1,104 @@
+require 'date'
+
+def random_username
+  "pl#{rand(999999).to_i}"
+end
+
+def change_password_manifest(username)
+  return <<-MANIFEST
+    user { '#{username}':
+      ensure   => 'present',
+      password => 'Password-#{rand(999999).to_i}'
+    }
+  MANIFEST
+end
+
+def assert_password_matches_on(host, username, password)
+  script = <<-PS1
+  Add-Type -AssemblyName System.DirectoryServices.AccountManagement
+  $ctx = New-Object System.DirectoryServices.AccountManagement.PrincipalContext("machine", $env:COMPUTERNAME)
+  $ctx.ValidateCredentials("#{username}", "#{password}")
+  PS1
+  execute_powershell_script_on(host, script) do |result|
+    assert_match(/True/, result.stdout, "Expected the the user's password not to be updated, but it was")
+  end
+end
+
+OLD_PASSWORD='0ldP@ssword'
+
+hosts.each do |host|
+  test_name 'PUP-6569 Passwords are not reset for disabled accounts' do
+    confine :to, platform: 'windows'
+
+    disabled_username = random_username
+
+    step "Create a disabled user account" do
+      on(host, "cmd.exe /c net user #{disabled_username} /active:no /add")
+    end
+
+    step "Try to change the disabled user account's password with puppet" do
+      apply_manifest_on(host, change_password_manifest(disabled_username)) do |result|
+        assert_match(/Warn/i, result.stderr, "Missing warning when attempting to reset a disabled user account's password") unless host['locale'] == 'ja'
+      end
+    end
+
+    step "Ensure the password wasn't changed" do
+      assert_password_matches_on(host, disabled_username, OLD_PASSWORD)
+    end
+
+    teardown do
+      on(host, "cmd.exe /c net user #{disabled_username} /delete", accept_all_exit_codes: true)
+    end
+  end
+
+  test_name 'PUP-6569 Passwords are not reset for expired accounts' do
+    confine :to, platform: 'windows'
+
+    expired_username = random_username
+
+    step "Create an expired user account" do
+      on(host, "cmd.exe /c net user #{expired_username} /expires:#{(Date.today - 1).strftime('%m-%d-%Y')} /add")
+    end
+
+    step "Try to change the expired user's password with puppet" do
+      apply_manifest_on(host, change_password_manifest(expired_username)) do |result|
+        assert_match(/Warn/i, result.stderr, "Missing warning when attempting to reset an expired user account's password") unless host['locale'] == 'ja'
```

Comment:
Might be useful to create an `assert_match` wrapper in `windows_utils.rb` that would abstract this out. Maybe `assert_match_en`? I think this should be `if host['locale'] == 'en'` in case Puppet decides to translate for other Windows boxes in the future (e.g. French).

Reply:
This locale special case was something I copied from other tests in the suite -- if we move it to a common module, I don't think it should be Windows specific (or even puppet-repo-specific), since localization applies on most platforms in most agent components. 

I would prefer some other way to check that a warning was printed - I didn't want to test the content of the message, since that's pretty brittle, but I wasn't sure how else to do it. I don't know that these warning tests are even that important.

Reply:
Agreed. At that point, I guess the `assert_match_en` wrapper should be added to Beaker.